pandas/core/internals/construction.py

```diff
@@ -171,44 +172,142 @@ def init_dict(data, index, columns, dtype=None):
     Segregate Series based on type and coerce into matrices.
     Needs to handle a lot of exceptional cases.
     """
-    if columns is not None:
-        from pandas.core.series import Series
-        arrays = Series(data, index=columns, dtype=object)
-        data_names = arrays.index
-
-        missing = arrays.isnull()
-        if index is None:
-            # GH10856
-            # raise ValueError if only scalars in dict
-            index = extract_index(arrays[~missing])
-        else:
-            index = ensure_index(index)
-
-        # no obvious "empty" int column
-        if missing.any() and not is_integer_dtype(dtype):
-            if dtype is None or np.issubdtype(dtype, np.flexible):
-                # GH#1783
-                nan_dtype = object
-            else:
-                nan_dtype = dtype
-            val = construct_1d_arraylike_from_scalar(np.nan, len(index),
-                                                     nan_dtype)
-            arrays.loc[missing] = [val] * missing.sum()
-
+    from pandas.core.series import Series
```

Comment:
this is way way too complicated

Reply:
...

Reply:
What is that comment supposed to achieve? I've put a ton of time into this. The DataFrame constructor is complicated with way too many special cases. I agree. This is what it takes to avoid the issue.

Reply:
Like come on? What was the point of that? Do you think I intentionally write complex code?

Reply:
oh course not
my point is that the perf fix is not worth it with this complexity